src/azure-cli/azure/cli/command_modules/keyvault/_validators.py

```diff
@@ -289,8 +289,8 @@ def validate_deleted_vault_or_hsm_name(cmd, ns):
     vault_name = getattr(ns, 'vault_name', None)
     hsm_name = getattr(ns, 'hsm_name', None)
 
-    if hsm_name:
-        raise InvalidArgumentValueError('Operation "purge" has not been supported for HSM.')
+    if hsm_name and 'keyvault recover' in cmd.name:
```

Comment:
A bug fix here. Run `az keyvault recover` will show error messgae `purge has not been supported for HSM`